documentation for list iterators

tobibeer opened the pull request:
Documentation for #1330...

* heavily modified **ListWidget**
    * extracted **NestedLists**
* added **IteratorVariables**

Jermolene commented on editions/tw5.com/tiddlers/demonstrations/NestedLists.tid:
````diff
@@ -0,0 +1,44 @@
+The list in the [[Types|$:/core/ui/MoreSideBar/Types]] tab is generated with this markup:
+
+```
+<$list filter="[!is[system]has[type]each[type]sort[type]]">
````
Here perhaps we could transclude the actual content, something like:

```
<pre>
<code>
<$text text={{$:/core/ui/MoreSideBar/Types}}/>
</code>
</pre>
```

tobibeer replied:
Shall we perhaps make a global wikitext helper macro of that for...

**$:/editions/tw5.com/wikitext-macros**

?

I'm not so fond of these macro names...

```
<<wikitext-example>>
<<wikitext-example-without-html>>
```

Perhaps make them...

```
<<example-html>>
<<example>>
```

?

And add...

```
<<example-code>>
```

Jermolene replied:
I think that the macro name `<<example>>` is too generic, better leave that for users own macros. I'd call this macro `<<plain-text tiddler>>`. Also I just realised that we should use the codeblock widget. That way the text will be syntax highlighted if the plugin is installed:

```
\define plain-text(tiddler)
<$codeblock language={{$tiddler$!!type}}>
<$text text={{$tiddler$}}/>
</$codeblock>
```

tobibeer replied:
Ok to add that to...

**$:/editions/tw5.com/wikitext-macros**

?

How about...

```
<<tw-demo>>
<<tw-demo-html>>
<<tw-code>>
<<tw-code-demo>>
```

?

Jermolene replied:
@tobibeer if we really want to change the names of those existing macros it's better to discuss that in a fresh ticket, not these line notes (they're not as persistent as comments on an issue).

tobibeer replied:
Ok, I'll just add the other macro for now.

tobibeer commented:
Ok, added the macro(s) and updated the examples. For now, it doesn't implement the codeblock widget as you suggested here https://github.com/Jermolene/TiddlyWiki5/pull/1332#discussion_r22762157 because it would only return `undefined`.

Jermolene commented:
> For now, it doesn't implement the codeblock widget as you suggested here #1332 (comment) because it would only return undefined.

I'd got the syntax wrong; it should be:

```
\define plain-text(tiddler)
<$codeblock language={{$tiddler$!!type}} code={{$tiddler$}}/>
```

tobibeer commented:
Ok, thanks. Updated.

Can you perhaps write a little doc tiddler on the codeblock widget?
Thant would be nice.

Jermolene commented:
Thanks @tobibeer 

> Can you perhaps write a little doc tiddler on the codeblock widget?

If you want to have a go at this while I'm away, have a look at this file:

https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/widgets/codeblock.js

You can also see how the codeblock widget is overridden by the highlight plugin:

https://github.com/Jermolene/TiddlyWiki5/blob/master/plugins/tiddlywiki/highlight/highlightblock.js

tobibeer commented:
Just so I understand how /prerelease works... how come **IteratorVariables** is not on there yet while other parts are? Just curious. I can imagine you're perhaps doing some rewording.

Jermolene commented:
@tobibeer I've pushed a new prerelease with the file extension fixed in f4005871b4d1b84baec848b19afc2a5bdd3511d6

tobibeer commented:
thx
